### ISSUE TITLE
`unused-function-argument` (`ARG001`) - false positive on pytest `hookspec` functions

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9803]
in pytest, a new hook function is defined using the `hookspec` decorator, and the function has no implementation as it's basically like an abstract method:

```py
from pytest import hookspec

@hookspec
def pytest_foo(value: int):  # error Unused function argument: `value`
    ...
```
[playground](https://play.ruff.rs/c8a25959-2278-4072-95ed-9894874dae5e)

ruff could detect the use of the `hookspec` decorator and not report the error here.

related:
- #7286 
- https://github.com/pytest-dev/pytest/issues/11300

**Comment by charliermarsh** [comment]
We should be able to support this. Can you link to any relevant docs on `hookspec`?

**Comment by DetachHead** [comment]
https://docs.pytest.org/en/7.1.x/how-to/writing_hook_functions.html#declaring-new-hooks

a couple things worth noting:
- the `hookspec` decorator isn't actually required to declare a hook. instead you have to hook `pytest_addhooks` and register the whole module:
  ```py
  # foo/hooks.py
  from pytest import hookspec
  
  def pytest_foo(value: int):
      ...
  ```
  ```py
  # conftest.py
  def pytest_addhooks(pluginmanager):
      from foo import hooks
  
      pluginmanager.add_hookspecs(hooks)
  ```
  as far as i can tell the `hookspec` decorator is mainly used to specify additional options to individual hooks (so it's not needed if you are just using the default options). however using the decorator without any options like in my original example is perfectly valid, and a good way to let ruff know it's a hookspec without needing multi-file analysis (#7447)
- pytest' plugin functionality comes from [pluggy](https://pluggy.readthedocs.io/en/stable/), so ideally ruff would also recognize hookspecs in any project that uses pluggy. however since decorators are defined like this, it would either need multi-file analysis to work, or just assume any decorator called `hookspec` is a pluggy hook:
  ```py
  hookspec = pluggy.HookspecMarker("myproject")
  hookimpl = pluggy.HookimplMarker("myproject")
  ```

**Comment by charliermarsh** [comment]
Assuming it's defined within the same file (which seems common), we _could_ detect that `@hookspec` resolves to `hookspec = HookspecMarker("...")`.

**Comment by DetachHead** [comment]
would still require special casing for pytest, since its `hookspec` decorators are defined in a different file to where they're used in plugins

**Comment by MichaReiser** [comment]
https://github.com/astral-sh/ruff/pull/12966
